broker.go

```diff
@@ -358,6 +372,33 @@ func (b *Broker) fetchMetadata(topics ...string) (*proto.MetadataResp, error) {
 		return resp, nil
 	}
 
+	var resp *proto.MetadataResp
+
+	ok := b.dialRun(func(conn *connection, addr string) bool {
```

Comment:
Would this not be trying to get metadata from the nodes that were already tried in the previous step? Is this step necessary?